samcli/commands/build/command.py

```diff
@@ -335,3 +370,42 @@ def _get_mode_value_from_envvar(name: str, choices: List[str]) -> Optional[str]:
         raise click.UsageError("Invalid value for 'mode': invalid choice: {}. (choose from {})".format(mode, choices))
 
     return mode
+
+
+def _process_env_var(container_env_vars: Optional[List[str]]) -> Dict:
```

Comment:
Do we use this format elsewhere? Is there any other command that has similar use case?

Reply:
nope, this is a unique pattern so far, but might have future use cases